makePoplogCommander.sh

```diff
@@ -476,46 +664,76 @@ void extendPath( char * prefix, char * path, char * suffix ) {
     free( buff );
 }
 
-void setUpEnvironment( char * base, int flags, Chain envv ) {
+****
+
+# Here we create three functions for setting the environment variables that
+# are unique to the build variants: nox, xm, xt. These will be called
+# nox_setUpEnvVars, xm_setUpEnvVars, xt_setUpEnvVars.
+VARIANTS='nox xt xm'
+for variant in $VARIANTS
+do
+    echo "void ${variant}_setUpEnvVars( char * base, bool inherit_env ) {"
+    cat _build/environments/$variant.env \
+    | sed -e 's/"/\\"/g' \
+    | sed -e 's/\([^=]*\)=\(.*\)/    setEnvReplacingUSEPOP( "\1", "\2", base, inherit_env );/'
```

Comment:
Shouldn't this be `s/\([^=]+\)=\(.*\)` instead? We want to have at least one char for the var name, though the value may be empty (so `*` is right here)

Reply:
Good spot!